dev/tools/dartdoc_checker.dart

```diff
@@ -0,0 +1,127 @@
+  if (!dartDocDir.existsSync()) {
+    throw Exception('Directory with dartdoc output (${dartDocDir.path}) does not exist.');
+  }
+
+  // Makes sure that the path we were given contains some of the expected
+  // libraries and HTML files.
+  final List<String> canaryLibraries = <String>[
+    'animation',
+    'cupertino',
+    'material',
+    'widgets',
+    'rendering',
+    'flutter_driver',
+  ];
+  final List<String> canaryFiles = <String>[
+    'Widget-class.html',
+    'Material-class.html',
+    'Canvas-class.html',
+  ];
+
+  print('Scanning for unresolved dartdoc directives...');
+
+  final List<FileSystemEntity> toScan = dartDocDir.listSync();
+  int count = 0;
+
+  while (toScan.isNotEmpty) {
+    final FileSystemEntity entity = toScan.removeLast();
+    if (entity is File) {
+      if (path.extension(entity.path) != '.html') {
+        continue;
+      }
+      canaryFiles.remove(path.basename(entity.path));
+
+      // TODO(goderbauer): Remove this exception when https://github.com/dart-lang/dartdoc/issues/2272 is fixed.
+      if (entity.path.endsWith('-class.html') || entity.path.endsWith('-library.html') ) {
+        continue;
+      }
+
+      count += _scanFile(entity);
+    } else if (entity is Directory) {
+      canaryLibraries.remove(path.basename(entity.path));
+      toScan.addAll(entity.listSync());
+    } else {
+      throw Exception('$entity is neither file nor directory.');
+    }
+  }
+
+  if (canaryLibraries.isNotEmpty) {
+    throw Exception('Did not find docs for the following libraries: ${canaryLibraries.join(', ')}.');
+  }
+  if (canaryFiles.isNotEmpty) {
+    throw Exception('Did not find docs for the following files: ${canaryFiles.join(', ')}.');
+  }
+  if (count > 0) {
+    throw Exception('Found $count unresolved dartdoc directives (see log above).');
+  }
+  print('No unresolved dartdoc directives detected.');
+}
+
+int _scanFile(File file) {
+  assert(path.extension(file.path) == 'html');
+  Iterable<String> matches = _pattern.allMatches(file.readAsStringSync())
+      .map((RegExpMatch m ) => m.group(0));
+
+  // TODO(goderbauer): Remove this exception when https://github.com/dart-lang/dartdoc/issues/1945 is fixed.
+  matches = matches
+      .where((String m) => m != '{@inject-html}')
+      .where((String m) => m != '{@end-inject-html}');
+
+  if (matches.isNotEmpty) {
+    stderr.writeln('Found unresolved dartdoc directives in ${file.path}:');
+    for (final String match in matches) {
+      stderr.writeln('  $match');
+    }
+  }
+  return matches.length;
+}
+
+// Matches all `{@` that are not within `<code></code>` sections.
+//
+// This regex may lead to false positives if the docs ever contain nested tags
+// inside <code> sections. Since we currently don't do that, doing the matching
+// with a regex is a lot faster than using an HTML parser to strip out the
+// <code> sections.
+final RegExp _pattern = RegExp('({@[^}\n]*}?)(?![^<>]*<\/code)');
```

Comment:
Added `r`.

Since it is essentially free, I do want to catch the error where a macro is not terminated (or has been terminated with a `)` instead of a `}` etc).

I don't fully understand the regex suggestion. Wouldn't that match things inside a code tag?

Reply:
Yes, I wasn't thinking about the regex in the right way: my suggestion would only match inside of a code block.  Please ignore it. :-)